src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java

```diff
@@ -58,10 +59,12 @@ private XdocUtil() {
      */
     public static Set<Path> getXdocsFilePaths() throws IOException {
         final Path directory = Paths.get(DIRECTORY_PATH);
-        try (DirectoryStream<Path> stream = Files.newDirectoryStream(directory, "*.xml")) {
+        try (Stream<Path> stream = Files.walk(directory)) {
             final Set<Path> xdocs = new HashSet<>();
-            for (Path entry : stream) {
-                xdocs.add(entry);
+            for (Path path : stream.collect(Collectors.toList())) {
+                if (Files.isRegularFile(path) && path.toString().endsWith(".xml")) {
+                    xdocs.add(path);
+                }
```

Comment:
`newDirectoryStream` isn't enough since we now have subdirectories. That's why I used `Files.walk()`

This fixes test 
```
[ERROR]   AllChecksTest.testAllCheckstyleModulesHaveXdocDocumentation:495->lambda$testAllCheckstyleModulesHaveXdocDocumentation$4:499 Module AnnotationUseStyle does not have xdoc documentation.
```